lib/resources/forms.js

```diff
@@ -30,9 +30,15 @@ const excelMimeTypes = {
   xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
 };
 
-const canReadForm = (auth, form) => (form.state === 'closed'
-  ? auth.canOrReject('form.read', form)
-  : auth.canOrReject(['open_form.read', 'form.read'], form));
+const canReadForm = (auth, form) => {
+  if (form.def.publishedAt === null) {
```

Comment:
I think I'd feel slightly more comfortable with `==` instead of `===`. We don't usually use `===` with `null`: we don't usually distinguish between `null` and `undefined`. I don't think it's impossible that we'll one day subtly change our database functions such that `form.def.publishedAt` becomes `undefined` instead of `null`. I can't think of much of a downside of checking for both `null` and `undefined`.

Reply:
:+1: fixed